### PR TITLE
add tests for /static and different file types

### PR DESCRIPTION
**Pull request by DavZim**
Adds tests for #180

### CODE REVIEW

File: inst/tinytest/test-app-content-handlers.R [hunk 1]
```diff
@@ -63,4 +63,21 @@ err_msg = paste0("lexical error: invalid char in json text.\n",
 expect_equal(rs[[1]], err_msg)
 expect_equal(rs[[2]], "text/plain")
 
+
+# Test static files
+files <- c("test.R", "test.txt", "testdata.csv", "testdata.rds", "testplot.jpg",
```
Comment (DavZim) [inline inst/tinytest/test-app-content-handlers.R:68]: Not sure if this is the best approach (hardcoding the files), or if we should get the files somewhere else.

File: inst/tinytest/test-app-content-handlers.R [hunk 2]
```diff
@@ -63,4 +63,21 @@ err_msg = paste0("lexical error: invalid char in json text.\n",
...
+rs = backend$convert_response(app$process_request(request_static_files))
+expect_equal(jsonlite::fromJSON(rs[[1]]), files)
+
+for (file in files) {
+  request_file = Request$new(path = paste0("/static/", file))
+  rs = backend$convert_response(app$process_request(request_file))
+  expect_true(grepl(paste0(file, "$"), rs[[1]]))
+  mime <- mime::guess_type(file)
+  expect_equal(rs[[2]], mime)
+}
```
Comment (DavZim) [inline inst/tinytest/test-app-content-handlers.R:80]: this doesn't test the file contents itself. Not sure if this needs to be tested or if we can assume that its ok?! :)

File: inst/examples/content-handlers/app.R [hunk 3]
```diff
@@ -5,6 +5,33 @@
...
+  plot(data$x, data$y, type = "l")
+  dev.off()
+})
+rm(a)
+
+writeLines("Hello World", file.path(static_dir, "test.txt"))
+writeLines("print('Hello World')", file.path(static_dir, "test.R"))
+write.csv(data, file.path(static_dir, "testdata.csv"))
+saveRDS(data, file.path(static_dir, "testdata.rds"))
+
```
Comment (DavZim) [inline inst/examples/content-handlers/app.R:34]: `data` and `static_dir` are not removed/cleared for now.  As this file is sourced, we might want to clear it?!

File: inst/examples/content-handlers/app.R [hunk 4]
```diff
@@ -47,6 +74,17 @@ app$add_post("/json", function(request, response) {
   response$body = serialize(request$body, NULL)
 })
 
+app$add_get("/csv", function(request, response) {
```
Comment (DavZim) [inline inst/examples/content-handlers/app.R:77]: not sure how this got in, tbh...

File: DESCRIPTION [hunk 5]
```diff
@@ -8,7 +8,7 @@ Description:
   application development. Out of the box allows to serve requests using
   'Rserve' package, but flexible enough to integrate with other HTTP servers
   such as 'httpuv'.
-Version: 0.4.1
+Version: 0.4.1001
```
Comment (DavZim) [inline DESCRIPTION:11]: Not sure how you handle versioning. Thought a version bump would be helpful. Feel free to change.

File: inst/tinytest/test-app-content-handlers.R [hunk 6]
```diff
@@ -63,4 +63,21 @@ err_msg = paste0("lexical error: invalid char in json text.\n",
 expect_equal(rs[[1]], err_msg)
 expect_equal(rs[[2]], "text/plain")
 
+
+# Test static files
+files <- c("test.R", "test.txt", "testdata.csv", "testdata.rds", "testplot.jpg",
+           "testplot.pdf", "testplot.png")
+request_static_files = Request$new(path = "/list_static_files")
+rs = backend$convert_response(app$process_request(request_static_files))
+expect_equal(jsonlite::fromJSON(rs[[1]]), files)
```
Comment (dselivanov) [inline inst/tinytest/test-app-content-handlers.R:72]: order of files in the `files` array and when we list files in the dir might be different. So we need to compare sets / sorted lists.

**Review by dselivanov — CHANGES_REQUESTED** [review]
looks good. Need to 
- address comment about file order
-  replace assignment operator from `<-` to `=`. We use `=` everywhere

**Comment by dselivanov** [comment]
Once this PR merged I plan to release 1.0.0 on the CRAN

File: inst/tinytest/test-app-content-handlers.R (hunk 6 above)
Comment (DavZim) [reply inst/tinytest/test-app-content-handlers.R]: This is handled by the changes in da1d045

**Comment by DavZim** [comment]
Using the new commits, all open issues should be addressed.

**Comment by DavZim** [comment]
Before the CRAN release. I planned on making a PR regarding Etags (see also https://stackoverflow.com/questions/71527627/r-restrserve-add-etag-to-static-path), do you have a place where we can discuss this? It would be great if that feature would make it to the next CRAN release as well.

**Comment by DavZim** [comment]
With PR #182, this is now independent from any discussion around ETags.

The requested changes have been made and this PR should be ready to be merged (at least from my perspective)

**Comment by dselivanov** [comment]
Thanks! Sorry for delay, just too many things on my plate this week. Will
merge soon.

**Pull request merged**